processing/src/main/java/org/apache/druid/frame/key/FrameComparisonWidgetImpl.java

```diff
@@ -56,34 +54,34 @@ public class FrameComparisonWidgetImpl implements FrameComparisonWidget
   private final Memory rowOffsetRegion;
   private final Memory dataRegion;
   private final int keyFieldCount;
-  private final List<KeyColumn> keyColumns;
   private final List<FieldReader> keyFieldReaders;
   private final int firstFieldPosition;
   private final RowKeyComparisonRunLengths rowKeyComparisonRunLengths;
-  // We memoize the serde instead of fetching it everytime from the global map, since that is thread-safe and is guarded by a
-  // ConcurrentHashMap, while we only access FrameComparisonWidget from a single thread.
-  private final Map<String, ComplexMetricSerde> serdeMap = new HashMap<>();
+  private final ComplexMetricSerde[] complexMetricSerdes;
```

Comment:
Could use a similar comment to the one I mentioned in ByteRowKeyComparator, to aid understanding.